### PR TITLE
update MutatingWebhook apiVersion to v1 and other minor fixes

### PR DESCRIPTION
**Pull request by tariq1890**
[X] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[X] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
This is introduced in k8s 1.16 right? We will need to disable our 1.15 tests if we are going to do this (which we should do - just want to make sure we do that before we merge this so we don't break the tests)

**Comment by tariq1890** [comment]
@howardjohn I don't think it will break. The `v1` allows backward compatibility. I believe that's what the `admissionReviewVersions` field is for

**Comment by howardjohn** [comment]
I don't think the v1 type exists in k8s 1.15 though, it was added in k8s
1.16 - can you verify?

**Comment by tariq1890** [comment]
@howardjohn Yes, you're right. As per the documentation, we need atleast k8s 1.16 to go forward with `v1`

File: tests/integration/security/webhook/config/sidecar-injector-webhook.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ metadata:
     release: istio
 webhooks:
   - name: sidecar-injector.istio.io
+    admissionReviewVersions: ["v1", "v1beta1"]
```
Comment (howardjohn) [inline tests/integration/security/webhook/config/sidecar-injector-webhook.yaml:12]: I am a bit confused how this works. Does this mean the api server will send admissionReview/v1 first, then fallback to v1beta1 if it fails or something? Why do we want both here?

**Review by howardjohn — COMMENTED** [review]
overall lgtm
@jasonwzm can you review?

File: manifests/charts/base/templates/validatingwebhookconfiguration.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: admissionregistration.k8s.io/v1beta1
+apiVersion: admissionregistration.k8s.io/v1
```
Comment (jasonwzm) [inline manifests/charts/base/templates/validatingwebhookconfiguration.yaml:1]: Is updating validatingwebhookconfiguration also in scope of this PR? Should we also update the code that uses admissionregistration v1beta1 package?

File: manifests/charts/istio-control/istio-discovery/Chart.yaml [hunk 3]
```diff
@@ -6,6 +6,8 @@ tillerVersion: ">=2.7.2"
 description: Helm chart for istio control plane
 keywords:
   - istio
+  - istiod
```
Comment (lei-tang) [inline manifests/charts/istio-control/istio-discovery/Chart.yaml:9]: Can you comment on why these two lines are needed?

File: manifests/charts/istiod-remote/files/gen-istiod-remote.yaml [hunk 4]
```diff
@@ -632,6 +632,8 @@ webhooks:
         namespace: istio-system
         path: "/inject"      
       caBundle: 
+    sideEffects: None
```
Comment (lei-tang) [inline manifests/charts/istiod-remote/files/gen-istiod-remote.yaml:635]: Can you comment on why the sideEffects is none for MutatingWebhookConfiguration?

File: manifests/charts/istiod-remote/templates/mutatingwebhook.yaml [hunk 5]
```diff
@@ -25,6 +25,8 @@ webhooks:
         path: "/inject"
       {{- end }}      
       caBundle: {{ .Values.sidecarInjectorWebhook.caBundle }}
+    sideEffects: None
```
Comment (lei-tang) [inline manifests/charts/istiod-remote/templates/mutatingwebhook.yaml:28]: Can you comment on why the sideEffects is none for MutatingWebhookConfiguration?

**Comment by nrjpoddar** [comment]
@tariq1890 what version of K8s was this migrated to v1?

**Comment by tariq1890** [comment]
@nrjpoddar K8s v1.16. The `v1beta1` will be removed from `v1.19` onwards

File: manifests/charts/istio-control/istio-discovery/Chart.yaml (hunk 3 above)
Comment (tariq1890) [reply manifests/charts/istio-control/istio-discovery/Chart.yaml:9]: They are added keywords to better the visibilty and searchability of the Chart

File: manifests/charts/istiod-remote/files/gen-istiod-remote.yaml (hunk 4 above)
Comment (tariq1890) [reply manifests/charts/istiod-remote/files/gen-istiod-remote.yaml:635]: It's a required field in apiVersion `admissionregistration/v1`

File: tests/integration/security/webhook/config/sidecar-injector-webhook.yaml (hunk 1 above)
Comment (hzxuzhonghu) [reply tests/integration/security/webhook/config/sidecar-injector-webhook.yaml:12]: >Does this mean the api server will send admissionReview/v1 first, then fallback to v1beta1 if it fails or something?

Yes, can be found in api docs

I think it is to be compatible with older webook which only serves v1beta1 admissionReview, currently we serve v1beta1 in istiod

**Comment by hzxuzhonghu** [comment]
>The v1beta1 will be removed from v1.19 onwards

Nope, it will be deprecated in k8s 1.19. So we don't have to bump so quicky as k8s 1.15 is widely used.  Most important point is that v1 did not bring many benefits.

**Comment by howardjohn** [comment]
@tariq1890 what is the status of this? Are you still working on it? Any blockers?

**Comment by hzxuzhonghu** [comment]
I would suggest postponing this pr unless we don't want to support k8s 1.15 anymore.

**Comment by howardjohn** [comment]
Istio 1.7 will support 1.16-1.18:
https://docs.google.com/document/d/14Xfe9V0vv8d7W75c21d-cY2XyK6xvRH0_Kg8j_zfh4Y/edit#bookmark=id.3gi4mzlz3xnp

**Comment by tariq1890** [comment]
If I get the OK to continue this, I can rebase and resume work on this PR.

**Comment by tariq1890** [comment]
@howardjohn Side note, can we have this [doc](https://docs.google.com/document/d/14Xfe9V0vv8d7W75c21d-cY2XyK6xvRH0_Kg8j_zfh4Y/edit#bookmark=id.3gi4mzlz3xnp) in istio.io docs? This would be a useful reference for Istio users.

**Comment by howardjohn** [comment]
We currently just list the current version support ,not past/future.
https://github.com/istio/istio.io/pull/7154 adds it but its been in limbo
(mostly my fault)

**Comment by howardjohn** [comment]
@tariq1890 are you still working on this? The code freezer for 1.7 is coming rapidly so we will need to get this in soon to make the release. Let me know if you don't have time and I can pick up the change

**Comment by tariq1890** [comment]
I can continue to iterate on this, as I believe most of the work has already been done. I am just trying to udnerstand the root cause of the e2e failures

**Comment by howardjohn** [comment]
Some look like its a merge conflict (fixed by rebase), 

`TestValidation/networking-v1beta-DestinationRule-invalid.yaml: validation_test.go:117: got unexpected success for invalid config` seems to imply the validation doesn't actually work anymore?

```
    TestOutboundTrafficPolicy_AllowAny_Mixer: helper.go:372: initialize instances: 2 errors occurred:
        	* timeout while waiting (last error: no pods found for [app=destination])
        	* timeout while waiting (last error: no pods found for [app=client])
```

suggests the mutating webhook also isn't applied properly

File: manifests/charts/base/templates/validatingwebhookconfiguration.yaml (hunk 2 above)
Comment (tariq1890) [reply manifests/charts/base/templates/validatingwebhookconfiguration.yaml:2]: Yes I have done that.

**Comment by tariq1890** [comment]
@howardjohn The fail test cases aren't too descriptive. Error messages here would be helpful to find what and where things went wrong here.

**Comment by tariq1890** [comment]
All the unit and non-integration tests pass. Looks like this is definitely missing a key piece of configuration. Can't seem to figure out what exactly though.

**Comment by howardjohn** [comment]
`TestValidation/networking-v1beta-DestinationRule-invalid.yaml: validation_test.go:115: got unexpected success for invalid config` seems fairly clear to me, I think the webhooks aren't actually doing anything

**Comment by tariq1890** [comment]
I am guessing for some reason, The `client` is not unable to pick up on the `v1` apiVersion of the ValidatingWebhookConfiguration

**Comment by tariq1890** [comment]
@howardjohn Any ideas on why the `v1` apiVersion is not being applied?

**Comment by tariq1890** [comment]
Requesting some help with this as I don't have an in-depth knowledge of the codebase and I don't have the bandwidth either to do the deep dive debugging. My OSS work is mostly voluntary and I have to prioritise most of my efforts and time towards my full-time job

**Comment by howardjohn** [comment]
The type needs to be fully specified it seems, meaning we add `TypeMeta: kubeApisMeta.TypeMeta{APIVersion: "admission.k8s.io/v1", Kind: "AdmissionReview"}` to the ` kubeApiAdmission.AdmissionReview` response

File: tests/integration/security/webhook/config/sidecar-injector-webhook.yaml (hunk 1 above)
Comment (tariq1890) [reply tests/integration/security/webhook/config/sidecar-injector-webhook.yaml:12]: Yup

**Comment by tariq1890** [comment]
Looks like the tests are all green now :)

**Review by howardjohn — APPROVED** [review]

**Review by ayj — APPROVED** [review]

**Review by JimmyCYJ — APPROVED** [review]

**Comment by nmittler** [comment]
@tariq1890 I suspect this may have broken things on GKE 1.16.13-gke.1. See #26186.

**Pull request merged**